test/integration/users_signup_test.rb

```diff
@@ -0,0 +1,16 @@
+require 'test_helper'
+
+class UsersSignupTest < ActionDispatch::IntegrationTest
+
+  test "invalid signup information" do
+    get signup_path
+    assert_no_difference 'User.count' do
+      post users_path, params: { user: { name:  "",
+                                         email: "user@invalid",
+                                         password:              "foo",
+                                         password_confirmation: "bar" } }
```

Comment:
```suggestion
     post users_path, params: {
        user: { name: '',
                email: 'user@invalid',
                password: 'foo',
                password_confirmation: 'bar' }
      }
```

少し見にくいのでインデントを調整してもいいかもしれませんね
（この通り修正せずに自分が見やすいと感じるスタイルでも大丈夫です！）

Reply:
わかりました！